### DISCUSSION TITLE
How to enable blur for only specific windows?

### DISCUSSION THREAD

**Discussion by xdevs23** [hyprwm/Hyprland#4169] [Q&A]
Most windows are opaque and some transparent windows may preferably not have a blurred background (like certain splash screens). I would like to only enable blur for specific classes, like the terminal and wofi but leave the rest without blur.

I have already tried using a windowrulev2 with noblur and negative lookahead in the regex but that did not do anything.

How can I achieve this?

**Comment by master3590** [comment]
`windowrule = noblur,^(firefox)$ # disables blur for firefox`
https://wiki.hyprland.org/Configuring/Window-Rules/

Sorry for the late reply, I was just researching about this and found the solution.
I hope it helps you out

**Reply by xdevs23** [reply]
Sadly this is not the solution as it only *disables* blur for specific cases but does not allow *enabling* blur for specific cases (only opt-out, not opt-in)

**Reply by cr4sh-me** [reply]
Maybe this is what you're looking for
`decoration {`
`  active_opacity = 1.0`
`  inactive_opacity = 1.0`
`  fullscreen_opacity = 1.0`
`...}`

**Comment by KoyeBosh** [comment]
You can use regex to match everything except the things you want to blur. Here is the snippet that I use:
```
windowrule = noblur,^(?!(neovide|foot|whatever else))
```
You can expand this rule to include as many programs as you'd like.

**Reply by xdevs23** [reply]
Can we do this using windowrulev2 ?

**Reply by 0xFMD** [reply]
yes, here's an examples

```
# this will remove the blur when visiting specific websites
windowrulev2 = noblur, title:^(.*(Twitch|YouTube|Plex|Udemy|Figma)).*(Firefox).*$ 

windowrulev2 = noblur, class:^(virt-manager)$,title:.*(on QEMU).*


```

**Comment by deepanchal** [comment]
I had the same issue and I had to enable `ignore_opacity` option in `decoration:blur` (https://wiki.hyprland.org/Configuring/Variables/#blur)

```hypr
blur {
  enabled = true
  size = 8
  passes = 2
  new_optimizations = yes
  ignore_opacity=true
}
```

Then, setting opacity for apps would also enable blur (ex. `opacity 0.8, initialTitle:^(Spotify.*)$`)

**Comment by 5urajkharkwal** [comment]


**Reply by Aqa-Ib** [reply]
you have to set the opacity through kitty's config file, not through hyprland's in order to achieve that.

**Reply by fufexan** [reply]
https://github.com/alexhulbert/Hyprchroma maybe this also helps.

**Reply by 5urajkharkwal** [reply]
Hyprland allows changing the opacity of everything in the terminal, like the Neovim status line and tmux status line, seamlessly blending them with the wallpaper for a visually appealing effect. In contrast, Kitty's opacity settings only affect the background, leaving other elements unchanged.

I've partially achieved this effect by combining Hyprland's opacity with Kitty's opacity and adding some blur effects. You can check out my dotfiles [here](https://github.com/SurajKharkwal/dotfiles-git) Although dotfiles are still not completed

**Comment by skyWatcherY** [comment]
I also faced this issue with mako. I find this thing helped a lot: https://wiki.hyprland.org/Configuring/Dispatchers/#setprop
so I use this code to just make mako blur : exec-once = [opacity 0.9] mako. 
"[...]" can use decorations(not for all ? I'm not sure)

**Comment by aruncs31s** [comment]
why cant you just this 

```bash
windowrule = opacity 0.7 override 0.7 override,class:^(thunar)$
```
this enables it just for thunar

**Comment by Nanito111** [comment]
this is my solution: `windowrule = noblur, class:negative:(foot$|zen$|thunar$)`

**edit**:
Hyprland uses Google's RE2 for RegEx parsing. This engine doesn't support negative lookup (eg. `?!(...)`). If you want to match everything that is not in the string, you can do it by putting the `negative:` prefix to the RegEx string.

**wiki**: https://wiki.hypr.land/0.52.0/Configuring/Window-Rules/#regex-writing
